angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts

```diff
@@ -0,0 +1,649 @@
+        >
+          <input
+            type="checkbox"
+            sprkSelectionInput
+            (click)="toggleSSNType()"
+            data-id="ssn-reveal-3"
+            id="show-ssn"
+          >
+          <label for="show-ssn" sprkSelectionLabel>Show SSN</label>
+        </sprk-selection-item-container>
+        <div
+          [hidden]="ssnInput.valid || ssnInput.pristine"
+          sprkFieldError
+        >
+          <sprk-icon
+            iconType="exclamation-filled-small"
+            additionalClasses="sprk-b-ErrorIcon"
+          >
+          </sprk-icon>
+          <div class="sprk-b-ErrorText">Invalid SSN.</div>
+        </div>
+    </sprk-input-container>
+  `,
+  props: {
+    ssnType: 'password',
+    toggleSSNType(): void {
+      this.ssnType === 'password'
+        ? (this.ssnType = 'text')
+        : (this.ssnType = 'password');
+    }
+  }
+});
+
+export const searchInput = () => ({
+  moduleMetadata: modules,
+  template: `
+    <sprk-input-container>
+      <label sprkLabel>Search Input</label>
+      <input
+      name="search_input"
+      type="search"
+      [(ngModel)]="search_input"
+      #searchInput="ngModel"
+      data-id="search-1"
+      sprkInput
+      >
+    </sprk-input-container>
+  `,
+});
+
+export const inlineSearchInput = () => ({
+  moduleMetadata: modules,
+  template: `
+    <sprk-icon-input-container>
+      <label
+        class="sprk-b-Label--with-icon sprk-u-ScreenReaderText"
+        sprkLabel
+      >
+        Search
+      </label>
+      <sprk-icon
+        iconType="search"
+        additionalClasses="sprk-b-InlineSearch__icon sprk-c-Icon--stroke-current-color"
+        sprk-input-icon
+      ></sprk-icon>
+      <input
+        name="inline_search_input"
+        class="sprk-b-TextInput--has-svg-icon"
+        type="text"
+        placeholder="Search"
+        [(ngModel)]="inline_search_input"
+        #inlineSearchInput="ngModel"
+        data-id="input-search-1"
+        sprkInput
+      >
+    </sprk-icon-input-container>
+  `,
+});
+
+
+export const monetaryInput = () => ({
+  moduleMetadata: modules,
+  template: `
+    <sprk-icon-input-container
+      iconContainerClasses="sprk-b-TextInputIconContainer--has-text-icon"
+    >
+      <label class="sprk-b-Label--monetary" sprkLabel> Payment </label>
+      <input
+        class="sprk-b-TextInput--has-text-icon"
+        name="monetary_input"
+        type="text"
+        pattern="(^\\$?(\\d+|\\d{1,3}(,\\d{3})*)(\\.\\d+)?$)|^$"
+        [(ngModel)]="monetary_input"
+        #monetaryInput="ngModel"
+        sprkFormatterMonetary
+        sprkInput
+      />
+      <div
+        [hidden]="monetaryInput.valid || monetaryInput.pristine"
+        sprkFieldError
+      >
+        <sprk-icon
+          iconType="exclamation-filled-small"
+          additionalClasses="sprk-b-ErrorIcon"
+        ></sprk-icon>
+        <div class="sprk-b-ErrorText">Invalid amount.</div>
```

Comment:
Depending on where we land with the period at the end of the SSN error, let's make sure this one matches.